File: code/ex107.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def traverse(self, root, level):
        global r
        
        if root == None:
            return
        
        if len(r) <= level:
            #Create a new subList for a new level
            r.insert(0, [])
            
        #Add value to the subList
        r[len(r) - level - 1].append(root.val)
        
        #Recursion
        self.traverse(root.left, level + 1)
        self.traverse(root.right, level + 1)
        
    
    def levelOrderBottom(self, root):
        """
        :type root: TreeNode
        :rtype: List[List[int]]
        """
        
        global r
        r = []
    
        self.traverse(root, 0)
    
        return r
        

File: code/ex107.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        if (root == NULL)    return {};
        
        vector<vector<int>> res;
        queue<TreeNode*> que;
        que.push(root);
        
        while (!que.empty()) {
            int size = que.size();
            vector<int> level;
            for (int z = size; z > 0; z--) {
                TreeNode* out = que.front();    que.pop();
                level.push_back(out->val);
                if (out->left != NULL)    que.push(out->left);
                if (out->right != NULL)    que.push(out->right);
            }
            res.push_back(level);
        }
        
        reverse(res.begin(), res.end());
        return res;
    }
};